pkg/agents/team5/agent.go

```diff
@@ -86,6 +93,114 @@ func (a *CustomAgent5) checkForLeader() {
 	if diceRoll >= a.leadership {
 		a.Log("An agent has become a leader", infra.Fields{"dice roll": diceRoll, "leadership": a.leadership, "selfishness": a.selfishness, "floor": a.Floor()})
 		//TODO: Send treaties here about eating less food
+		hpLevel := a.currentAimHP - ((a.currentAimHP-a.HealthInfo().WeakLevel)/10)*(diceRoll-a.leadership)
+		a.currentProposal = messages.NewTreaty(messages.HP, hpLevel, messages.LeavePercentFood, 100, messages.GE, messages.EQ, 5, a.ID())
+		a.treatySendCounter = 1
+		a.Log("Agent is sending a treaty proposal", infra.Fields{"Proposed Max Hp": hpLevel})
+	}
+}
+
+func (a *CustomAgent5) statementsIntersect(op1 messages.Op, value1 int, op2 messages.Op, value2 int) bool {
+	switch op1 {
+	case messages.EQ:
+		return ((op2 == messages.EQ && value1 == value2) ||
+			(op2 == messages.LT && value1 < value2) ||
+			(op2 == messages.LE && value1 <= value2) ||
+			(op2 == messages.GT && value1 > value2) ||
+			(op2 == messages.GE && value1 >= value2))
+	case messages.LT:
+		return ((op2 == messages.EQ && value1 < value2) ||
```

Comment:
Please correct me if I'm getting this wrong. If the first condition is to be less than value1, and the second is to be equal to value 2, they would intersect if value1 > value 2 wouldn't they???

Reply:
yes, I will flip these!